src/app/attendance/components/AttendanceFormBuilder.tsx

```diff
@@ -12,15 +12,19 @@ import {
   FormControl,
   FormMessage,
 } from "@/components/ui/form";
-import { useEffect } from "react";
+import React from "react";
 import { createAttendance } from "../actions/create-attendance";
 import { toast } from "@/hooks/use-toast";
 import FormWrapper from "@/common/FormWrapper";
 import { MemberSelector } from "@/common/MemberSelector/MemberSelector";
+import { MemberRole } from "@/constant/constant";
+import { updateAttendance } from "../actions/update-attendance";
 import { User } from "@/types";
-import { MemberRole, UserStatus } from "@/constant/constant";
-import { getAttendanceById } from "../actions/getAttendanceById";
-import { useParams } from "next/navigation";
+
+interface AttendanceFormBuilderProps {
+  initialData?: AttendanceFormValues;
+  loginUser?: User;
```

Comment:
I don't think `loginUser` should be optional. Also, @muhammad-bilal-ali-khokhar created a hook called `userUser`. We should be getting it from here.

Reply:
Fixed